atd-vzv/src/views/map/InfoBox/infoBoxStyles.js

```diff
@@ -14,6 +17,28 @@ export const StyledMobileInfo = styled.div`
   .card {
     background: none;
     border: none;
-    max-width: ${responsive.drawerWidth - 20}px;
+    max-width: ${maxInfoBoxWidth}px;
   }
 `;
+
+export const setPopupPosition = (popupX) => {
+  // Use the max width of info box to decide if it can fit centered on map point
+  // or if it needs to offset from right or left of viewport
+  const halfInfoBoxWidth = maxInfoBoxWidth / 2;
+  const canInfoBoxFitCentered =
+    popupX - halfInfoBoxWidth > 0 &&
+    popupX + halfInfoBoxWidth < window.innerWidth;
+  const isInfoBoxTooFarLeft = popupX - halfInfoBoxWidth <= 0;
+  const isInfoBoxTooFarRight = popupX + halfInfoBoxWidth >= window.innerWidth;
+
+  switch (!!popupX) {
+    case canInfoBoxFitCentered:
+      return `left: ${popupX - halfInfoBoxWidth};`;
+    case isInfoBoxTooFarLeft:
+      return `left: ${halfInfoBoxWidth - popupX}px;`;
+    case isInfoBoxTooFarRight:
+      return `right: ${popupX + halfInfoBoxWidth - window.innerWidth}px;`;
+    default:
+      return null;
+  }
+};
```

Comment:
It works great! One thing I am not sure of is wether it needs to control the y-axis too. But clicking around the x-axis seems to be well taken care of.

This is what it looks like when I click on a low element within the browser:
<img width="1680" alt="2020-06-29_09-52-10" src="https://user-images.githubusercontent.com/5282430/86022019-6cafbc00-b9ef-11ea-90dc-a147bb2d5b4c.png">

Reply:
Great point, Sergio. I talked with Jace about this, and he thinks handling the x-axis only is good for now. Forgot to mention that in the PR. Thanks for taking a look at this in BrowserStack! 📱